examples/dashboard/src/components/ServerSdkContext.tsx

```diff
@@ -26,33 +32,77 @@ type Props = {
 };
 
 export const ServerSDKProvider = ({ children }: Props) => {
-  const [serverAddress, setServerAddressState] = useState<string>(() => {
-    const serverAddress = localStorage.getItem(localStorageServerAddress);
-    return serverAddress ? serverAddress : "localhost:4000";
-  });
-
-  const setServerAddress = useCallback(
-    (value: string) => {
-      setServerAddressState(value);
-      localStorage.setItem(localStorageServerAddress, value);
-    },
-    [setServerAddressState]
+  const [host, setHost] = useLocalStorageStateString(LOCAL_STORAGE_HOST_KEY, "localhost:5002");
+  const [protocol, setProtocol] = useLocalStorageStateString(LOCAL_STORAGE_PROTOCOL_KEY, "wss");
+  const [path, setPath] = useLocalStorageStateString(LOCAL_STORAGE_PATH_KEY, "/socket/peer/websocket");
+  const [isSecure, setIsSecure] = useLocalStorageStateString(LOCAL_STORAGE_PROTOCOL_KEY, "wss");
+
+  const [signalingWebsocket, setSignalingWebsocket] = useState<string | null>(null);
+  const [serverMessagesWebsocket, setServerMessagesWebsocket] = useState<string | null>(null);
+  const [httpApiUrl, setHttpApiUrl] = useState<string | null>(null);
+
+  const [serverToken, setServerToken] = useLocalStorageStateString("serverToken", "development");
+
+  const setHostInput = useCallback((value: string) => {
+    setHost(value);
+    localStorage.setItem(LOCAL_STORAGE_HOST_KEY, value);
+  }, []);
+
+  const setProtocolInput = useCallback((value: string) => {
+    setProtocol(value);
+    localStorage.setItem(LOCAL_STORAGE_PROTOCOL_KEY, value);
+  }, []);
+
+  const setPathInput = useCallback((value: string) => {
+    setPath(value);
+    localStorage.setItem(LOCAL_STORAGE_PATH_KEY, value);
+  }, []);
+
+  useEffect(() => {
+    const restProtocol = protocol === "wss" ? "https" : "http";
+
+    const abc = `${restProtocol}://${host}`;
+    console.log({ abc });
+    setHttpApiUrl(abc);
+  }, [host, protocol]);
+
+  const client = useMemo(
+    () =>
+      axios.create({
+        headers: {
+          Authorization: `Bearer ${serverToken}`,
+        },
+      }),
+    [serverToken]
   );
-  const roomApi = useMemo(() => new RoomApi(undefined, `http://${serverAddress}`, client), [serverAddress]);
-  const peerApi = useMemo(() => new PeerApi(undefined, `http://${serverAddress}`, client), [serverAddress]);
 
-  const peerWebsocket: string = useMemo(() => serverAddress, [serverAddress]);
-  const serverWebsocket: string = useMemo(() => `ws://"${peerWebsocket}/socket/server/websocket`, [peerWebsocket]);
+  const roomApi = useMemo(
+    () => (httpApiUrl ? new RoomApi(undefined, httpApiUrl || "", client) : null),
```

Comment:
I don't think that they add anything here, but that's just my preference (the same goes for `peerApi` below). 
```suggestion
    () => httpApiUrl ? new RoomApi(undefined, httpApiUrl || "", client) : null,
```

Reply:
I've implemented your suggestion, and Prettier has reformatted it